packages/cli/src/workflows/workflows.controller.ee.ts

```diff
@@ -13,3 +18,44 @@ EEWorkflowController.use((req, res, next) => {
 	// use ee router
 	next();
 });
+
+/**
+ * (EE) PUT /workflows/:id/share
+ *
+ * Grant or remove users' access to a workflow.
+ */
+
+EEWorkflowController.put('/:workflowId/share', async (req: WorkflowRequest.Share, res) => {
+	const { workflowId } = req.params;
+	const { shareWithIds } = req.body;
+
+	if (!Array.isArray(shareWithIds) || !shareWithIds.every((userId) => typeof userId === 'string')) {
+		return res.status(400).send('Bad Request');
+	}
+
+	const { ownsWorkflow, workflow } = await EEWorkflows.isOwned(req.user, workflowId);
+
+	if (!ownsWorkflow || !workflow) {
+		return res.status(403).send();
+	}
+
+	let newShareeIds: string[] = [];
+	await Db.transaction(async (trx) => {
+		// remove all sharings that are not supposed to exist anymore
+		await EEWorkflows.pruneSharings(trx, workflowId, [req.user.id, ...shareWithIds]);
+
+		const sharings = await EEWorkflows.getSharings(trx, workflowId);
+
+		// extract the new sharings that need to be added
+		newShareeIds = rightDiff(
+			[sharings, (sharing) => sharing.userId],
+			[shareWithIds, (shareeId) => shareeId],
+		);
+
+		if (newShareeIds.length) {
+			await EEWorkflows.share(trx, workflow, newShareeIds);
```

Comment:
I am not sure but I think we do, since we want the transaction to finish after all queries execute. Might not be necessary but it won't hurt since we await the top level transaction as well.